requirements.txt

```diff
@@ -4,3 +4,6 @@ cython==0.29.33
 reno==3.3.0  # for changelog
 
 setuptools>=46.4.0  # to support setup.cfg getting __version__
+
+networkx
```

Comment:
Unfortunately I don't think we want to add NetworkX as a direct dependency of dimod just for these methods. It doesn't look like any of NetworkX's algorithms are used so probably we can replace the use of the Graph class internally.

We do provide some support for functions accepting graphs if NetworkX is installed, see https://github.com/dwavesystems/dimod/blob/8433f221a1e79101e1db0d80968ab5a2f59b865d/dimod/decorators.py#L228 but we use more generic structured internally so we don't require the user have it installed.